ddsketch/store/collapsing_lowest_dense_store.go

```diff
@@ -102,7 +102,7 @@ func (s *CollapsingLowestDenseStore) extendRange(newMinIndex, newMaxIndex int) {
 func (s *CollapsingLowestDenseStore) adjust(newMinIndex, newMaxIndex int) {
 	if newMaxIndex-newMinIndex+1 > len(s.bins) {
 		// The range of indices is too wide, buckets of lowest indices need to be collapsed.
-		newMinIndex = newMaxIndex - len(s.bins) + 1
+		newMinIndex = newMaxIndex - len(s.bins)
```

Comment:
It's not clear to me why we'd want this. Can you elaborate? Can you also add a failing test case that illustrates the unexpected behavior that  motivates the change?

If we go with this change, we need to adjust [`CollapsingHighestDenseStore`](https://github.com/DataDog/sketches-go/blob/master/ddsketch/store/collapsing_highest_dense_store.go#L100) as well.

Reply:
I opened the PR during an incident. I should have opened a draft. This is actually not the issue. The library is crashing though with when it receives sketches with infinite max:
```goroutine 850 [running]:
github.com/DataDog/sketches-go/ddsketch/store.(*CollapsingLowestDenseStore).adjust(0xc00ecb5ae0, 0x0?, 0x4d3bbc?)
	/go/pkg/mod/github.com/!data!dog/sketches-go@v1.3.0/ddsketch/store/collapsing_lowest_dense_store.go:118 +0x1c5
github.com/DataDog/sketches-go/ddsketch/store.(*CollapsingLowestDenseStore).extendRange(0xc00ecb5ae0, 0x42e20d?, 0x43d4038?)
	/go/pkg/mod/github.com/!data!dog/sketches-go@v1.3.0/ddsketch/store/collapsing_lowest_dense_store.go:85 +0x14a
github.com/DataDog/sketches-go/ddsketch/store.(*CollapsingLowestDenseStore).normalize(0xc00ecb5ae0, 0x8000000000000000)
	/go/pkg/mod/github.com/!data!dog/sketches-go@v1.3.0/ddsketch/store/collapsing_lowest_dense_store.go:61 +0x36
github.com/DataDog/sketches-go/ddsketch/store.(*CollapsingLowestDenseStore).AddWithCount(0xc00ecb5ae0, 0x7ff0000000000000?, 0xfff8000000000000)
	/go/pkg/mod/github.com/!data!dog/sketches-go@v1.3.0/ddsketch/store/collapsing_lowest_dense_store.go:50 +0x34
github.com/DataDog/sketches-go/ddsketch.changeStoreMapping.func1(0x7fffffff, 0x3f847ae147ae147b)
	/go/pkg/mod/github.com/!data!dog/sketches-go@v1.3.0/ddsketch/ddsketch.go:456 +0x1fb
github.com/DataDog/sketches-go/ddsketch/store.(*DenseStore).ForEach(0xc00ece2f00, 0xc011f7b0c0)
	/go/pkg/mod/github.com/!data!dog/sketches-go@v1.3.0/ddsketch/store/dense_store.go:204 +0x66
github.com/DataDog/sketches-go/ddsketch.changeStoreMapping({0x2c5d380?, 0xc002611230}, {0x2c5d380?, 0xc001df7fb0}, {0x2c63ec0, 0xc00ece2f00}, {0x2c63e20?, 0xc00ecb5ae0}, 0x3ff0000000000000)
	/go/pkg/mod/github.com/!data!dog/sketches-go@v1.3.0/ddsketch/ddsketch.go:445 +0x10f
github.com/DataDog/sketches-go/ddsketch.(*DDSketch).ChangeMapping(0xc00ece2f80, {0x2c5d380?, 0xc001df7fb0}, {0x2c63e20?, 0xc00ecb5ae0}, {0x2c63e20?, 0xc00ecb5b30}, 0x3ff0000000000000)
	/go/pkg/mod/github.com/!data!dog/sketches-go@v1.3.0/ddsketch/ddsketch.go:437 +0xdb
github.com/DataDog/dd-go/trace/apps/data-pipeline-resolver.apmToMetricsSketch(0x0?, 0xc00ece2f80)
	/go/src/github.com/DataDog/dd-go/trace/apps/data-pipeline-resolver/extractor.go:337 +0x29a
github.com/DataDog/dd-go/trace/apps/data-pipeline-resolver.(*extractor).toPoints(0xc00eb87110?, {{0xc0156ca500, 0x2, 0x2}, {0xc0023ee0b6, 0x3}, {0xc0023ee0d0, 0xa}, {0xc002ebf440, 0x20}, ...})
	/go/src/github.com/DataDog/dd-go/trace/apps/data-pipeline-resolver/extractor.go:241 +0x605
github.com/DataDog/dd-go/trace/apps/data-pipeline-resolver.(*extractor).run(0xc00eb87110)
	/go/src/github.com/DataDog/dd-go/trace/apps/data-pipeline-resolver/extractor.go:100 +0x185
created by github.com/DataDog/dd-go/trace/apps/data-pipeline-resolver.(*extractor).start
	/go/src/github.com/DataDog/dd-go/trace/apps/data-pipeline-resolver/extractor.go:87 +0x45
```
I wasn't able to reproduce in tests

Reply:
I will close the PR, and re-open it if I can reproduce.